### PR TITLE
Pasting into Existing columns latest

### LINKED ISSUE #5619: Adding a simple paste into a data frame
As I write the help it would now be very useful if we could have a paste into the data frame. 

Previously we have been hesitant, because pasting is complicated to do well.

I don't want to spend long on the pasting, partly because I think we may want to use a new grid control in the future.  But also because it will take forever to do well.  That's why I am suggesting just a single column (variable) at a time.  You have quoted one day for this work and I suggest that this is a good limit (at least for now.)  What can you do (if anything) on the pasting of a single column within a day?   

So, I suggest a very simple paste would be useful and consistent with our current use of R-Instat.  We can (I think) already paste a single variable into the levels/labels dialogue and perhaps elsewhere.  Could we similarly also paste a single variable (column) into a data sheet.  Usually it will be into one with just missing (NAs) currently.  If so, then it simply pastes.  If too short, then some NAs remain (top or bottom).  If too long, then it could give a message, or just paste what it can.
a) If the variable is already there, then it could give a warning.
b) It only pastes down - never across.  One small enhancement is therefore that if the data in the clipboard goes across, then it still works and pastes down.
c) Of course if you can do more, particularly pasting into multiple columns, that's fine.  And it could be as limited as you like, e.g. only when they are empty, or only if they are all numeric, or ...

### PR DESCRIPTION
**Pull request by Patowhiz**
Fixes #5619

Please not this PR replaces PR #5982 and #6334 but does all that is explained in those PRs.

Pasting into existing columns is only supported if the data pasted is of the same length(which is dependent on the pasting position) as the data frame, and the pasting the column header is not supported.

A user can paste into the entire column(in this case the data pasted should be of equal length to the data frame selected) or paste from a particular cell(in this case the data pasted should be equal to the remainder of the number of rows from the selected cell row).

@africanmathsinitiative/developers this is ready for review.

### CODE REVIEW

**Comment by Patowhiz** [comment]
@dannyparsons in this PR, I have factored all the items we discussed on skype. After your review I can go ahead and now do pasting new columns and new data frame.

**Review by rdstern — APPROVED** [review]
@Patowhiz this seems to work well from my point of view.
@lloyddewit over to you.

**Comment by Patowhiz** [comment]
@lloyddewit @dannyparsons I'll be glad if this could be reviewed. It's blocking me from making 2 new PRs.

**Comment by maxwellfundi** [comment]
@lloyddewit Could you please review this?

**Review by dannyparsons — APPROVED** [review]

File: instat/ucrDataView.vb [hunk 1]
```diff
@@ -984,4 +998,79 @@ Public Class ucrDataView
...
+    ''' <param name="startRowPos">starting row position. This starts at postion 1</param>
+    ''' <param name="firstClipRowHeader">flag indicating whether first row of clipboard data is a header</param>
+    Private Sub PasteValuesToDataFrame(lstColumnNames As IEnumerable(Of String), startRowPos As String, firstClipRowHeader As Boolean)
+        Dim clsPasteValues As New RFunction
+        Dim strColNamesVec As String = ""
+        Dim strClipBoardText As String = My.Computer.Clipboard.GetText
+
+        If String.IsNullOrEmpty(strClipBoardText) Then
+            MsgBox("No data available for pasting.", MsgBoxStyle.Information, "No Data")
+            Exit Sub
```
Comment (shadrackkibet) [inline instat/ucrDataView.vb:1040]: If the clipboard is empty I think it is better to disable paste and therefore we don't need this message? That is what is done in other software such as excel and word? In Rstudio paste is enabled but it does nothing!

File: instat/ucrDataView.vb [hunk 2]
```diff
@@ -984,4 +998,79 @@ Public Class ucrDataView
...
+        Next
+        strColNamesVec = "c(" & strColNamesVec & ")"
+
+        'add R parameters and run the command
+        clsPasteValues.SetRCommand(frmMain.clsRLink.strInstatDataObject & "$paste_from_clipboard")
+        clsPasteValues.AddParameter("data_name", Chr(34) & grdCurrSheet.Name & Chr(34))
+        clsPasteValues.AddParameter("col_names", strColNamesVec)
+        clsPasteValues.AddParameter("start_row_pos", startRowPos)
+        clsPasteValues.AddParameter("first_clip_row_is_header", If(firstClipRowHeader, "TRUE", "FALSE"))
+        clsPasteValues.AddParameter("clip_board_text", Chr(34) & strClipBoardText & Chr(34))
```
Comment (shadrackkibet) [inline instat/ucrDataView.vb:1064]: parameter postions?

**Review by shadrackkibet — COMMENTED** [review]
- This should be added into column metadata right-click. The right clicks are not consistent anymore.
- We have paste under the Edit menu and the icon on the toolbar. I think it should be enabled now.
- Since we now have paste on right-click, we need copy as well?

I think we can explore ways to avoid right-click being long in some instances. We can decide to hide these features in some instances.

**Pull request merged**